apps/system/js/browser_settings.js

```diff
@@ -46,6 +46,11 @@
         request.onsuccess = function() {
           request.result.clearBrowserData();
         };
+
+        // Clear the icon dataStore.
+        LazyLoader.load('shared/js/icons_helper.js').then(() => {
+          IconsHelper.clear();
```

Comment:
We need to not clear the icons for pinned sites and pinned pages, which is going to be a bit tricky. Not sure there's anything we can do in this bug.